csrc/scheduler/utils.cpp

```diff
@@ -783,24 +797,23 @@ ReductionTvProperties getReductionProperties(
   int64_t inner_most_dimension_numel = 1;
   int64_t inner_most_dimension_ndims = 0;
 
+  ExpressionEvaluator& expr_eval = runtime_info.expressionEvaluator();
+
   // Start from the inner most dimension, and work outwards. If this is a 3D
   // pattern, i.e. theres a pattern like [r0, r1, i2, r3] or [i0, r1, r2, i3,
   // i4] then compute the inner most dimension to compute separately.
-  const auto& root_dom = tv->getMaybeRootDomain();
-  for (size_t i = root_dom.size(); i > 0; i--) {
-    auto id = root_dom[i - 1];
+  const auto& logical_domain = tv->getLogicalDomain();
```

Comment:
I wonder why not just look at the allocation domain.

@jjsjann123 Would the old code work if the allocation domain was reordered?

Reply:
I don't think reduction scheduler handles allocation domain properly. We had a conversation before about that: https://github.com/NVIDIA/Fuser/issues/2202#issuecomment-2301992369

Back to your question, I don't think the old code is right. We should really be counting the element number based on their allocation domain. But in order to do that, the scheduler needs to first schedule the loop domain with a re-order to match the loop and allocation.

For this PR, I think the changes here looks right to support multi device split/sharding. But it's not fixing the old problem on reduction scheduler's support of allocation domain.